### PR TITLE
3ds2 updates

### PR DESCRIPTION
**Pull request by mbretter**
handle orders without shipping addresses

### CODE REVIEW

File: app/code/community/WirecardEE/PaymentGateway/Helper/Threeds.php [hunk 1]
```diff
@@ -126,6 +126,10 @@ public function getCardCreationDate($tokenId)
      */
     public function getAddressFirstUsed($addressId)
     {
+        if (is_null($addressId)) {
+            return null;
+        }
+
         /** @var Mage_Sales_Model_Resource_Order_Collection $orderCollection */
         $orderCollection = Mage::getModel('sales/order')
```
Comment (serucee) [inline app/code/community/WirecardEE/PaymentGateway/Helper/Threeds.php]: This call is used 3-4 times, maybe move to a method "getMageOrderCollection"

File: app/code/community/WirecardEE/PaymentGateway/controllers/GatewayController.php [hunk 2]
```diff
@@ -219,7 +229,7 @@ protected function buildOrderSummary(Mage_Sales_Model_Order $order, PaymentInter
                 $threedsHelper->getCustomerLastLogin(),
                 $threedsHelper->getChallengeIndicator(),
                 $threedsHelper->isNewToken($tokenId),
-                $threedsHelper->getAddressFirstUsed($order->getShippingAddress()->getCustomerAddressId()),
+                $threedsHelper->getAddressFirstUsed($shippingAddressId),
```
Comment (serucee) [inline app/code/community/WirecardEE/PaymentGateway/controllers/GatewayController.php]: Maybe add a check in getAddressFirstUsed that returns before going through the whole method with a null as address id

File: app/code/community/WirecardEE/PaymentGateway/Helper/Threeds.php (hunk 1 above)
Comment (mbretter) [reply app/code/community/WirecardEE/PaymentGateway/Helper/Threeds.php]: done

File: app/code/community/WirecardEE/PaymentGateway/controllers/GatewayController.php (hunk 2 above)
Comment (mbretter) [reply app/code/community/WirecardEE/PaymentGateway/controllers/GatewayController.php]: getAddressFirstUsed has already a null-check

**Pull request merged**